### PR TITLE
feat(trigger): add a new prop "containerScrollToClose" supporting close popup when scrolling

### PR DESCRIPTION
**Pull request by yulitao-byte**
## Types of changes

- [x] New feature
- [ ] Bug fix
- [ ] Enhancement
- [ ] Documentation change
- [ ] Coding style change
- [ ] Component style change
- [ ] Refactoring
- [ ] Test cases
- [ ] Continuous integration
- [ ] Typescript definition change
- [ ] Breaking change
- [ ] Others 

## Background and context

Trigger组件在多个嵌套滚动容器情况下，会有popup被遮挡，或者popup浮出当前滚动容器等一些edge case，因此，增加一个closeOnScroll属性，用于简化处理。
另外，和UX沟通，一般用户滚动的话，可能这个时候是不需要看菜单的，如有这样的情况，便可以使用该属性。

## Changelog

| Component | Changelog(CN) | Changelog(EN) | Related issues |
| --------- | ------------- | ------------- | -------------- |
|      Trigger     |     `Trigger` 组件支持通过 `containerScrollToClose` 属性设置弹出层挂载容器滚动时，关闭弹出层。           |   The `Trigger` component supports the popup layer to be closed when the container is scrolled by setting the `containerScrollToClose` property.            |                |

## Checklist:

- [ ] Test suite passes (`npm run test`)
- [ ] Provide changelog for relevant changes (e.g. bug fixes and new features) if applicable.
- [ ] Changes are submitted to the appropriate branch (e.g. features should be submitted to `feature` branch and others should be submitted to `main` branch)

### CODE REVIEW

File: components/Trigger/index.tsx [hunk 1]
```diff
@@ -226,8 +227,14 @@ class Trigger extends PureComponent<TriggerProps, TriggerState> {
...
-      this.onContainersScroll();
+    if (currentProps.updateOnScroll || currentProps.closeOnScroll) {
+      this.onContainersScroll(
+        currentProps.closeOnScroll
+          ? () => {
+              this.setPopupVisible(false);
+            }
+          : this.handleUpdatePosition
+      );
     }
```
Comment (yinkaihui) [inline components/Trigger/index.tsx:238]: 解绑事件需要处理下。 要不直接把 ```currentProps.closeOnScroll
          ? () => {
              this.setPopupVisible(false);
            }``` 的逻辑放到 `handleScroll` 里，在 handleScroll 里去进行逻辑判断吧

Comment (yulitao-byte) [reply components/Trigger/index.tsx:232]: Done 烦请再review下

**Comment by yulitao-byte** [comment]
修改prop名称为 containerScrollToClose

**Review by yinkaihui — APPROVED** [review]
